src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedLanguageServerDefinition.java

```diff
@@ -157,9 +171,19 @@ public Object getLanguageServerInitializationOptions() {
         return initializationOptions;
     }
 
+    public Map<String, ClientConfigurationLanguageSettings> getLanguageServerClientConfiguration() {
+        if ((clientConfiguration == null) && (clientConfigurationContent != null) && !clientConfigurationContent.isBlank()) {
+            try {
+                clientConfiguration = JSONUtils.getLsp4jGson().fromJson(clientConfigurationContent, ClientConfigurationLanguageSettings.TYPE_TOKEN);

```

Comment:
Unlike the server-side settings and initialization options, we have a concrete schema here and can deserialize into a specific type, in this case, a map keyed by language ID with values that are per-language client-side settings.